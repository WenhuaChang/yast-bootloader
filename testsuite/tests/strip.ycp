/**
 * File:
 *      strip.ycp
 *
 * Module:
 *      Bootloader
 *
 * Summary:
 *      testsuite for striping strings
 *
 * Authors:
 *      Jiri Srain <jsrain@suse.cz>
 *      primary author unknown
 *
 * $Id$
 *
 */

{
    import "Pkg";
    include "testsuite.ycp";

    map READ_I = $[
        "target" : $[
            "size" : -1,
            "tmpdir" : "/tmp/",
	    "yast2" : nil,
        ],
        "probe" : $[
            "architecture" : "i386",
            "has_apm" : true,
            "has_pcmcia" : false,
            "has_smp" : false,
            "system" : [],
            "memory" : [],
            "cpu" : [],
            "cdrom" : $[
                "manual" : [],
            ],
            "floppy" : $[
                "manual" : [],
            ],
        ],
        "sysconfig" : $[
            "console" : $[
                "CONSOLE_FONT" : "",
                "CONSOLE_SCREENMAP" : "",
                "CONSOLE_UNICODEMAP" : "",
                "CONSOLE_MAGIC" : "",
                "CONSOLE_ENCODING" : "",
            ],
            "language" : $[
                "RC_LANG" : "",
                "DEFAULT_LANGUAGE" : "",
            ],
        ],
        "etc" : $[
            "install_inf" : $[
                "Cmdline" : "",
            ],
        ],
        "proc" : $[
            "cpuinfo" : $[
                "value" : $[
                    "0" : $[
                        "flags" : ""
                    ],
                ],
            ],
        ]
    ];

    TESTSUITE_INIT ([READ_I], 0);
    include "bootloader/routines/misc.ycp";
    TESTSUITE_CLEAN ();

    TEST(``(sformat("a\nb\nc")), [], nil);

    TEST(``(strip("")), [], nil);
    TEST(``(strip(" ")), [], nil);
    TEST(``(strip("a")), [], nil);
    TEST(``(strip("    a")), [], nil);
    TEST(``(strip("a    ")), [], nil);
    TEST(``(strip("a   a")), [], nil);
    TEST(``(strip("   a ")), [], nil);
    TEST(``(sformat("%1", [strip(" \t  a ")])), [], nil);
    TEST(``(sformat("%1", [strip(" \n   \n\t a\n  a\n ")])), [], nil);
    TEST(``(sformat("%1", [strip("\t\t\t    s \t a\n ")])), [], nil);
}
