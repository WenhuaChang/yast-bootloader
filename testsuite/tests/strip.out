Read	.target.size "/etc/install.inf" -1
Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Read	.proc.cpuinfo.value.0."flags" ""
Read	.probe.memory []
Read	.probe.cpu []
Read	.probe.floppy.manual []
Read	.probe.cdrom.manual []
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Execute	.target.bash_output "locale -k charmap" $[]
Return	a
Return	
Return	
Return	a
Return	a
Return	a
Return	a   a
Return	a
Return	["a"]
Return	["a\n  a"]
Return	["s \t a"]
